src/ext/ddtrace_extra.c

```diff
@@ -0,0 +1,118 @@
+#ifdef HAVE_CONFIG_H
```

Comment:
What is the motivation for moving these functions out of **ddtrace.c**? The name of the file **ddtrace_extra.c** isn't obvious to me what it contains and all other PHP extensions I've seen in the wild keep all the PHP API stuff in their "dedicated to PHP" file. So unless there's really good reason to break this out, I'm -1 for this separate file. Just my 2 cents. But if we decide to move forward with this, we'll need to add this to the **package.xml** file. :)

Reply:
Motivation is to not to have everything in one file :) 
There are many extensions that separate PHP things to different files (like uopz), php itself is doing this as well. 

My rationale here is to cutout most things not directly related to tracing to separate file. The `_extra` is only meant as a conversation starter :) - I'm entirely open to different naming/ separation schemes.

If you feel that its more important to keep everything in one file thats cool too. I Just didn't want to add yet another thing to file that contain so many components of the extension already.

Reply:
For sure! I'm all about separating things out into smaller files to make things much easier to maintain. My comment is more about how we break it up. To use the example of **ext/uopz**, I love how [the main **uopz.c** file](https://github.com/krakjoe/uopz/blob/master/uopz.c) has just the PHP API stuff in it - INI settings, PHP info, PHP_FUNCTION's, and ZPP. All the implementation details are in separate files.

That's what I'm advocating here - just keeping all the parts tying the extension into PHP in the **ddtrace.c** file so there's one "source of truth" for all the PHP API stuff. And then we make sure we've separating any specific implementation logic in separate files. What do you think? :)

Reply:
Indeed, my memory was a bit fuzzy there it seems :D lets keep it in one file then - for now at least :)  its still quite manageable at this moment